seatunnel-apis/seatunnel-api-spark/src/main/java/org/apache/seatunnel/spark/batch/SparkBatchExecution.java

```diff
@@ -0,0 +1,132 @@
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+import java.util.List;
+
+public class SparkBatchExecution implements Execution<SparkBatchSource, BaseSparkTransform, SparkBatchSink> {
+
+    public static final String SOURCE_TABLE_NAME = "source_table_name";
+    public static final String RESULT_TABLE_NAME = "result_table_name";
+
+    public static void registerTempView(String tableName, Dataset<Row> ds) {
+        ds.createOrReplaceGlobalTempView(tableName);
+    }
+
+    public static void registerInputTempView(BaseSparkSource<Dataset<Row>> source, SparkEnvironment environment) {
+        Config config = source.getConfig();
+        if (config.hasPath(SparkBatchExecution.RESULT_TABLE_NAME)) {
+            String tableName = config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(tableName, source.getData(environment));
+        } else {
+            throw new ConfigRuntimeException(
+                    "Plugin[" + source.getClass().getName() + "] must be registered as dataset/table, please set \"result_table_name\" config");
+        }
+    }
+
+    public static Dataset<Row> transformProcess(SparkEnvironment environment, BaseSparkTransform transform, Dataset<Row> ds) {
+        Dataset<Row> fromDs;
+        Config config = transform.getConfig();
+        if (config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME)) {
+            String sourceTableName = config.getString(SparkBatchExecution.SOURCE_TABLE_NAME);
+            fromDs = environment.getSparkSession().read().table(sourceTableName);
+        } else {
+            fromDs = ds;
+        }
+        return transform.process(fromDs, environment);
+    }
+
+    public static void registerTransformTempView(BaseSparkTransform transform, Dataset<Row> ds) {
+        Config config = transform.getConfig();
+        if (config.hasPath(SparkBatchExecution.RESULT_TABLE_NAME)) {
+            config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(RESULT_TABLE_NAME, ds);
+        }
+    }
+
+    public static void sinkProcess(SparkEnvironment environment, BaseSparkSink<?> sink, Dataset<Row> ds) {
+        Dataset<Row> fromDs;
+        Config config = sink.getConfig();
+        if (config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME)) {
+            String sourceTableName = config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            fromDs = environment.getSparkSession().read().table(sourceTableName);
+        } else {
+            fromDs = ds;
+        }
+        sink.output(fromDs, environment);
+    }
+
+    private final SparkEnvironment environment;
+
+    private Config config = ConfigFactory.empty();
+
+    public SparkBatchExecution(SparkEnvironment environment) {
+        this.environment = environment;
+    }
```

Comment:
better to moved to the top of the class.

Reply:
[7402da4](https://github.com/apache/incubator-seatunnel/pull/1188/commits/7402da4b0359459912acdd72bf7714491bf89f99) had been resolved